include/z64audio.h

```diff
@@ -16,7 +16,18 @@
 #define ADSR_GOTO -2
 #define ADSR_RESTART -3
 
-#define AIBUF_LEN 0x580
+// size of a single sample point
+#define SAMPLE_SIZE sizeof(s16)
+
+// Samples are processed in groups of 16 called a "frame"
+#define SAMPLES_PER_FRAME 16
```

Comment:
There's a define in abi.h for this, `ADPCMFSIZE`

Reply:
Oh nice! Is `ADPCMFSIZE` based on official documentation, or could we mess with the name a little (i.e. `ADPCMFSIZE` -> `ADPCM_FSIZE`)? Also, the `FSIZE` is "frame-size" right? I'm also noticing `ADPCMVSIZE`, what's the V there? `v-size`?

Reply:
`ADPCMFSIZE` is kind of an unfortunate name actually because it uses "SIZE" in its name for the number of samples, where I'm really trying to use "size" exclusively for "bytes" since it's easy to mix up these different units

Reply:
`ADPCMFSIZE` is from the original abi.h. FSIZE is indeed "frame size", I'm not 100% sure what VSIZE is but I assume "vector size".
Yeah, it is unfortunate that it clashes. What I'd like ultimately is that any numbers the microcode should also be aware of should have their defines in abi.h, we can be flexible with exactly how we achieve that since we know for sure that abi.h is custom anyway so editing it further is fine imo

Reply:
vector-size does make sense since the rsp does process samples in batches of 8 iirc. I ended up going with:
`#define SAMPLES_PER_FRAME ADPCMFSIZE` which I'm not sure is better, it still uses our custom-named macro while still tying it to the original abi.h name

Reply:
@Thar0 is this considered solved?

Reply:
Adding onto this concept, I've been testing reverb lately to understand docs and the define `RESAMPLE_STATE` in `abi.h` could be applied for a few of the unknown reverb states:
```
reverb->unk_30 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, sizeof(RESAMPLE_STATE));
reverb->unk_34 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, sizeof(RESAMPLE_STATE));
reverb->unk_38 = AudioHeap_AllocZeroed(&gAudioContext.miscPool, sizeof(RESAMPLE_STATE));
reverb->unk_3C = AudioHeap_AllocZeroed(&gAudioContext.miscPool, sizeof(RESAMPLE_STATE));
```
which might be more appropriate (addresses the comment below as well), but not sure if I should address that here or in reverb doc PRs when these unknowns are properly doc'd

Reply:
And related, there's also a filter state which hasn't been defined which probably should be, which would be more appropriate for a lot of the filtering heap allocations. But that still hasn't been introduced, so there's a bit more clean-up which is needed and seems like it could be done separate? Would be curious as to tharo's thoughts.